chassis/core/src/main/java/com/griddynamics/jagger/master/HibernateSessionIdProvider.java

```diff
@@ -23,20 +23,20 @@
 import java.util.List;
 
 import org.springframework.orm.hibernate3.HibernateTemplate;
+import com.griddynamics.jagger.exception.TechnicalException;
 import org.springframework.orm.hibernate3.support.HibernateDaoSupport;
 
-import com.griddynamics.jagger.exception.TechnicalException;
+import java.util.List;
 
 public class HibernateSessionIdProvider extends HibernateDaoSupport implements SessionIdProvider {
-	private final Object lock = new Object();
 	private String sessionId;
     private String sessionName;
     private String sessionComment;
 
 	@Override
 	public String getSessionId() {
 		if (sessionId == null) {
-			synchronized (lock) {
+			synchronized (this) {
```

Comment:
It looks like internal object lock pattern, which is safer than synchronizing on this - maybe author knew something and did it intentionally... I don't think it's bad, just wondering why it was used and will your change break something.